### PR TITLE
feat: AI text to speech for proposal body

### PR DESCRIPTION
**Pull request by wa0x6e**
# AI text to speech 

Generate a .mp3 audio file through OpenAI API from the proposal's body 

NOTE:

- Will work only for body with at most 4096 chars (limit on OpenAI side), and at least 500 chars
- Audio files are cached and saved on AWS after first call

## Test 

- Ensure you have `OPENAI_API_KEY` env var set in your `.env`
- (optional) In `.env`, set `STORAGE_ENGINE` to `file`, to save the mp3 file in `/tmp` locally, for easier access
- Send a curl POST request to: `http://localhost:3005/api/ai/tts/[PROPOSAL-ID]` (choose a proposal with a body length between 500 and 4096 chars)
- It should return a Buffer for an audio file (may take some time if file is not cached yet)
- If you have set your storage engine to file, go to /tmp to find and listen to the audio file

### CODE REVIEW

**Comment by wa0x6e** [comment]
For reviewers, check the approved sister PR https://github.com/snapshot-labs/snapshot-sidekick/pull/240 for an idea on the class structure and flow

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: .env.example [hunk 1]
```diff
@@ -9,6 +9,9 @@ AWS_BUCKET_NAME=
 WEBHOOK_AUTH_TOKEN=
 STORAGE_ENGINE=file
 VOTE_REPORT_SUBDIR=votes
+AI_SUMMARY_SUBDIR=ai-summary
+AI_TTS_SUBDIR=ai-tts
```
Comment (ChaituVR) [inline .env.example:13]: Could we also add default values for these, if someone forget. also not changing in `.env.test`

Comment (wa0x6e) [reply .env.example:13]: Default value is `undefined`, as defined here

https://github.com/snapshot-labs/snapshot-sidekick/blob/61a39b511596a7396c8aee2386f09fc99d529b9b/src/helpers/utils.ts#L41

**Comment by ChaituVR** [comment]
Somehow i always get `RECORD_NOT_FOUND` when i go to http://localhost:3005/api/ai/tts/0x770ff4e02634c77aaa09952345551168920f7878b32ab03fcef92763a5fb70ab

Any idea?

**Comment by wa0x6e** [comment]
> Somehow i always get `RECORD_NOT_FOUND` when i go to http://localhost:3005/api/ai/tts/0x770ff4e02634c77aaa09952345551168920f7878b32ab03fcef92763a5fb70ab
> 
> Any idea?

Are you sending a POST request ?

File: src/api.ts [hunk 2]
```diff
@@ -54,9 +55,38 @@ router.post('/ai/summary/:id', async (req, res) => {
...
     return rpcSuccess(res.status(200), summary, id);
-  } catch (e) {
+  } catch (e: any) {
     capture(e);
-    return rpcError(res, 'INTERNAL_ERROR', id);
+    return rpcError(res, e.message || 'INTERNAL_ERROR', id);
+  }
+});
+
+router.post('/ai/tts/:id', async (req, res) => {
```
Comment (ChaituVR) [inline src/api.ts:64]: Any reason to use `post` here instead of `get` and for summery when we are not using any input from `body`

Comment (wa0x6e) [reply src/api.ts:64]: Very basic bot protection, to avoid link prefetch or any other script/tools calling this API endpoint

**Review by ChaituVR — APPROVED** [review]
tAck

But i think we should check requests are coming from snapshot.org
and also have some rate limit on these routes

**Pull request merged**